client/landing/gutenboarding/path.ts

```diff
@@ -96,3 +104,32 @@ export function useCurrentStep() {
 export function useNewQueryParam() {
 	return new URLSearchParams( useLocation().search ).has( 'new' );
 }
+
+export function useIsAnchorFm(): boolean {
+	const podcastId = useAnchorFmPodcastId();
+	return podcastId !== null && podcastId !== '';
```

Comment:
```suggestion
	return Boolean(useAnchorFmPodcastId());
```

Reply:
I'm a little worried that this might create some strange behaviors when people give wrong anchorFM ids. I'd check if they match a certain pattern. Because any non-empty string can pass this check and throw Gutenboarding flow off.

Reply:
Added validation here: [add/custom-anchor-gutenboarding-steps 499dc09c4e] Validate podcast ID